Add  new config under `probe.driver.check` to ignore some IDE errors (#207)
VirtusLab/ide-probe: docs/driver.md

By default, the driver waits `30 seconds` for the IDE to connect the probe. After exceeding this time, the whole test fails.  
#### Automatic checks
```
driver.check {
    errors = false
    freezes = false

>     ignoreErrorsWithMessageContaining = []

Reviewer: Hmm I see both `snakeCase` and `kebab-case` conventions for HOCON in the wild, but I'd say kebab-case is more prevalent... consider reconfiguring pureconfig library if it expects `snakeCase` (although my impression was that `kebab-case` was the pureconfig's default 🤔)

Author: now I change configuration that `kebab-case ` need to be in configuration file and `snakeCase` (as it was configured before) in scala case class

}
```

By default, the driver doesn't fail the test upon detecting any errors or freezes during the execution.

> Field `ignoreErrorsWithMessageContaining` is an empty list by default. Putting some error messages into this field will cause probe to ignore IDE errors with these messages. 

Reviewer: ```suggestion
Field `ignoreErrorsWithMessageContaining` is an empty list by default. Putting some strings into this field will cause probe to ignore IDE errors whose messages contain any of the given strings. 
```

By default, the driver waits `30 seconds` for the IDE to connect the probe. After exceeding this time, the whole test fails.  
#### Automatic checks
```
driver.check {
     errors {
       enabled = false
       exclude-messages = []

>        include-messages = [".*"]

Author: I added `-messages` because include is special keyword in HOCON and could not use it (means It is possible but need to wrap `include` in additional double quotes)

Reviewer: Wow I did know that `include` is a special keyword but didn't realize it's forbidden even as a key 😯

Reviewer: (unless escaped, yeah)

Author: I know that `include-messages` looks worse than simply `include`, but probably `include-messages` is better than `"include"`

Reviewer 2: Alternatively "includes" and "excludes" or "included" and "excluded"

     }

>     freezes = false

Reviewer: While you're at this, to prevent introducing yet another breaking release in future... pls change so that `freezes` is an object with `enabled` field as well. The other fields may (or may not) be added in future (but without breaking anything)

>     freezes = false

Reviewer: Pls update

}
```

By default, the driver doesn't fail the test upon detecting any errors or freezes during the execution.

> Field `exclude-messages` is an empty list by default. Putting some regular expressions into this field will cause probe to ignore IDE errors whose messages match any of the given expressions, even if `probe.driver.check.errors.enabled` is `true`.

Reviewer: As usual problem with regexes all of the industry... is it:
1. _whose messages **fully** match any of the given expressions_ or
2. _whose messages **contain a substring** matching any of the given expressions_ or
(I.e. _anchored_ vs _unanchored_ regex)

Pls make it clear from the docs... for this purpose, I think 2. is more natural (so that `.*` at the start/end can be avoided)

Reviewer 2: Especially that you rather won't be fully matching a whole stacktrace :p
